pkg/agent/externalnode/external_node_controller.go

```diff
@@ -0,0 +1,753 @@
+	return nil
+}
+
+func (c *ExternalNodeController) deleteExternalNode(key string) error {
+	klog.InfoS("Deleting ExternalNode", "key", key)
+	obj, exists, _ := c.syncedExternalNodes.GetByKey(key)
+	if !exists {
+		klog.InfoS("Skipping ExternalNode deletion as it hasn't been synced", "ExternalEntityKey", key)
+		return nil
+	}
+	en := obj.(*v1alpha1.ExternalNode)
+	ifName, err := getHostInterfaceName(en.Spec.Interfaces[0])
+	if err != nil {
+		return err
+	}
+	if err := c.deleteInterface(ifName); err != nil {
+		return err
+	}
+	c.syncedExternalNodes.Delete(en)
+	return nil
+}
+
+func (c *ExternalNodeController) deleteInterface(ifName string) error {
+	hostIface, portExists := c.ifaceStore.GetInterfaceByName(ifName)
+	if !portExists {
+		klog.InfoS("Skipping deleting host interface since it doesn't exist ", "ifName", ifName)
+		return nil
+	}
+	klog.InfoS("Deleting interface", "ifName", ifName)
+	if err := c.removeOVSPortsAndFlows(hostIface); err != nil {
+		return err
+	}
+	c.ifaceStore.DeleteInterface(hostIface)
+	return nil
+}
+
+func (c *ExternalNodeController) createOVSPortsAndFlows(uplinkName, hostIfName, eeNamespace string, eeName string, ips []string) (*interfacestore.InterfaceConfig, error) {
+	adapterConfig, err := util.GetUplinkConfig(hostIfName)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get the configuration on the host interface", "hostInterface", hostIfName)
+		return nil, err
+	}
+	if err := util.RenameInterface(hostIfName, uplinkName); err != nil {
+		klog.ErrorS(err, "Failed to rename host interface name as the uplink name", "hostInterface", hostIfName, "uplink", uplinkName)
+		return nil, err
+	}
+	success := false
+	defer func() {
+		if !success {
+			util.RenameInterface(uplinkName, hostIfName)
+		}
+	}()
+	// Create uplink port on OVS.
+	uplinkExternalIDs := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaUplink,
+	}
+	uplinkUUID, err := c.ovsBridgeClient.CreatePort(uplinkName, uplinkName, uplinkExternalIDs)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create uplink port on OVS", "uplink", uplinkName)
+		return nil, err
+	}
+	defer func() {
+		if !success {
+			c.ovsBridgeClient.DeletePort(uplinkUUID)
+		}
+	}()
+	uplinkOFPort, err := c.ovsBridgeClient.GetOFPort(uplinkName, false)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get uplink ofport", "uplink", uplinkName)
+		return nil, err
+	}
+	klog.V(2).InfoS("Added uplink port on OVS", "port", uplinkName)
+
+	// Create host port on OVS.
+	hostIfUUID, err := c.ovsBridgeClient.CreateInternalPort(hostIfName, 0, nil)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create host port on OVS", "hostInterface", hostIfName)
+		return nil, err
+	}
+	defer func() {
+		if !success {
+			c.ovsBridgeClient.DeletePort(hostIfUUID)
+		}
+	}()
+	hostOFPort, err := c.ovsBridgeClient.GetOFPort(hostIfName, false)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get host interface ofport", "hostInterface", uplinkName)
```

Comment:
klog.ErrorS(err, "Failed to get host interface ofport", "hostInterface", hostIfName)